src/main/java/com/rultor/agents/github/PreviousReleases.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2009-2015, rultor.com
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the rultor.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rultor.agents.github;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.jcabi.aspects.Immutable;
+import com.jcabi.github.Release;
+import com.jcabi.github.Releases;
+import com.jcabi.log.Logger;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+
+/**
+ * Encapsulates a set of previous releases and a currently proposed release.
+ *
+ * @author Jimmy Spivey (JimDeanSpivey@gmail.com)
+ * @version $Id$
+ * @since 1.57
+ */
+@Immutable
+public final class PreviousReleases {
+
+    /**
+     * The proposed tag.
+     */
+    private final transient Version version;
+
+    /**
+     * Previously released tags.
+     */
+    private final transient Releases priors;
+
+    /**
+     * Default constructor.
+     * @param ver The proposed tag, wrapped in a Version object
+     * @param previous A list of previously released tags.
+     */
+    public PreviousReleases(final Version ver, final Releases previous) {
+        this.version = ver;
+        this.priors = previous;
+    }
+
+    /**
+     * Is this tagged release valid.
+     * @return True if the release is valid
+     */
+    public boolean isTagValid() {
```

Comment:
@JimDeanSpivey maybe rename this to `isValid`  only? I would also add to the javadoc what a valid tag is...